doc/development-good-practice.md

```diff
@@ -0,0 +1,50 @@
+# Git Hooks Management
+
+## Overview
+
+**Git Hooks**: A trigger to automate the checks before operations like commit, push and etc. 
+
+Our Git hooks are designed to check for proper formatting, linting, and test checking for no-pull-request-breakage. Using these Git hooks would ease the development and would be a good practice.
+
+Once the git hooks are set up in the clone repository/repositories in your local.
+
+*__Functionality of pre-commit__*: Whenever you try to `git commit` the pre-commit hook will execute to process before committing.
+Your commit will be committed but you will be able to notice the warning for the formatting, and linting flaws in your commit. 
+
+**NOTE**: To by-pass the pre-commit test, use either of the commands:
+* `THOTH_NO_PRECOMMIT_TEST=1 git commit -m "<commit msg>"`
+* `git commit -m "<commit msg>" --no-verify"` 
+
+
+*__Functionality of pre-push__*: Whenever you try to `git push` the pre-push hook will execute to process before pushing.
+Your push request will be executed but you will be able to notice the warning for the test passing flaws in your commit.
+
+**NOTE**: To by-pass the pre-commit test, use either of the commands:
+* `THOTH_NO_PREPUSH_TEST=1 git push origin master`
+* `git push origin master --no-verify"` 
+
+
+## Install Git hooks on local
+
+Our Git hooks are stored under the directory *.git_init/hooks*.
```

Comment:
@goern @harshad16 

I dont see that directory. 
1. I would also link to the git repo via [org/repo/git_init](org/repo/git_init)
2. Dont hide the init directory via a leading dot. Move it into doc or a template directory.

Sorry to comment on a closed PR.